azure/scope/machinepool.go

```diff
@@ -153,13 +152,13 @@ func (m *MachinePoolScope) Name() string {
 	return m.AzureMachinePool.Name
 }
 
-// ProviderID returns the AzureMachinePool ID by parsing Spec.FakeProviderID.
+// ProviderID returns the AzureMachinePool ID by parsing Spec.ProviderID.
 func (m *MachinePoolScope) ProviderID() string {
-	parsed, err := noderefutil.NewProviderID(m.AzureMachinePool.Spec.ProviderID)
+	resourceID, err := azure.ParseResourceID(m.AzureMachinePool.Spec.ProviderID)
 	if err != nil {
 		return ""
 	}
-	return parsed.ID()
+	return resourceID.Name
```

Comment:
is resourceID.Name() the equivalent of what we had before here? Seems that we're returning the instance ID (which itself seems strange since the function name implies it returns the provider ID of the resource.

Let's add a small unit test that passes before the change to validate before and after is equivalent?

Reply:
Comparing the code behind each, `resourceID.Name` should be equivalent, but I agree it deserves a unit test. Will do.

Reply:
I added a unit test for `MachinePoolScope.ProviderID()` against the `main` branch, then cherry-picked it into this branch to see that it continues to pass after the reimplementation. 👍🏻